zk/conn.go

```diff
@@ -476,6 +520,18 @@ func (c *Conn) loop() {
 			wg.Add(1)
 			go func() {
 				<-reauthChan
+				// This condition exists for signaling purposes, that the test
+				// `TestRecurringReAuthHang` was successful. The previous call
+				// `<-reauthChan` was not blocking. That means the
+				// `resendZkAuth` didn't block even on read loop error.
+				// See `TestRecurringReAuthHang`
+				if c.shouldDebugCloseRecvLoop() {
+					select {
+					case <-c.debugReauthDone:
```

Comment:
why do we try to read from this channel here?

The test is reading from it as well, in what it seems a wait for `close(c.debugReauthDone)` which happens in the `default:` case below. Can we drop the `select` statement altogether?